packages/react-router-dev/vite/plugin.ts

```diff
@@ -128,11 +129,11 @@ const CLIENT_ROUTE_EXPORTS = [
   "shouldRevalidate",
 ];
 
-/** This is used to manage a build optimization to remove unused route exports
-from the client build output. This is important in cases where custom route
-exports are only ever used on the server. Without this optimization we can't
-tree-shake any unused custom exports because routes are entry points. */
-const BUILD_CLIENT_ROUTE_QUERY_STRING = "?__remix-build-client-route";
+const ROUTE_ENTRY_QUERY_STRING = "?__react-router-entry=1";
```

Comment:
the query string can be prettier ala `?entry=true` or something since it doesn't affect URLs, just route module imports